### DISCUSSION TITLE
read-only-indicator : deprecation warning on pulsar

### DISCUSSION THREAD

**Discussion by stefaniecg** [pulsar-edit/.github#176] [Community Package Help]
The **read-only-indicator** pulsar package [1-2] is not working on pulsar. 

It says an item was _deprecated_, the error it shows is:
`Use customElements.define instead of document.registerElement see https://javascript.info/custom-elements`

Does anyone know how to fix this?

Thank you for the help you can provide.

Ref
https://web.pulsar-edit.dev/packages/read-only-indicator
https://github.com/alefragnani/atom-read-only-indicator

**Comment by Daeraxa** [comment]
As far as I know that should only be a warning but it is something that would need to be changed in the code but the maintainer has decided to archive their repo.

I tried installing it and it works fine for me, it adds the icon to the bottom left of my screen in the status bar.

**Comment by stefaniecg** [comment]
ok, thank you for your feedback.  

How can I fix this deprecation warning that it gives me?
does anyone know?

**Reply by confused-Techie** [reply]
To fix the deprecation you'd need to modify the source code of the package. But like @Daeraxa said if the package is archived you won't be able to fix that exact package.

But if you'd like, you can fork the package, fix the deprecations, then publish the package to the Pulsar Package Registry under a new name.

If you do decide to do that be sure to let us know!

**Reply by stefaniecg** [reply]
Can i fix it on my own local files? 
How can i know in which files is the problem? 
Where are the package files for pulsar locaded?

**Reply by confused-Techie** [reply]
You absolutely could just fix it for yourself locally if you don't want to publish the package.

But probably the best start would be to search for `document.registerElement` within the package itself, to see where it pops up, as it likely occurs in multiple locations.

Beyond that the location of Pulsar's downloaded packages will vary from OS. But generally will be located in `~/.pulsar/packages` (or Windows `%userprofile%/.pulsar/packages`)

**Reply by savetheclocktower** [reply]
Just so you know: that deprecation notice is not something that needs to be fixed. Deprecations are just a way of saying “this will stop working at some point in the future.” Lots, lots, lots of Pulsar packages use that old API. If it ever stopped working in Pulsar, we'd have to do a lot of work to make sure those packages didn't break.

**Reply by stefaniecg** [reply]
Thank you all for your feedack. 
I see now that the warning is not critical and the package will still work. 
The deprecation notice is anoying though... :)

**Reply by savetheclocktower** [reply]
You can search your own packages for the package called `deprecation-cop` and disable it. That should remove the notice.
